pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -851,7 +864,18 @@ func getInstallerPodName(ns *operatorv1.NodeStatus) string {
 }
 
 // ensureInstallerPod creates the installer pod with the secrets required to if it does not exist already
-func (c *InstallerController) ensureInstallerPod(ctx context.Context, operatorSpec *operatorv1.StaticPodOperatorSpec, ns *operatorv1.NodeStatus) error {
+// returns whether or not requeue and if an error happened while creating installer pod
+func (c *InstallerController) ensureInstallerPod(ctx context.Context, operatorSpec *operatorv1.StaticPodOperatorSpec, ns *operatorv1.NodeStatus) (bool, error) {
+	// checks if new revision should be rolled out
+	if c.shouldRevisionInstall != nil {
```

Comment:
I think this check should be moved to `manageInstallationPods` because this function is supposed to be only about creating the installer pod.

Reply:
we could maybe even move it inside the `sync` function before the call to `manageInstallationPods` since we want to skip the installation process entirely when the preconditions are not fulfilled.

Reply:
yeah we thought so, but it's better to continue the current rollout. See https://github.com/openshift/library-go/pull/1749#discussion_r1642499291

Reply:
good point. it made me notice that we might be subject to the following issue: https://github.com/openshift/library-go/pull/1749#discussion_r1662686144